### DISCUSSION TITLE
WAR Deployment

### DISCUSSION THREAD

**Discussion by tgarif** [dokku/dokku#5045] [General]
https://devcenter.heroku.com/articles/war-deployment

Does dokku has this plugins to deploy java WAR file?

**Comment by josegonzalez** [comment]
We do not have anything that supports a WAR file directly. The problem is we don't have a server that can host app slugs (which are essentially built containers) so you'd need that. My gut says that your best bet is to host the WAR file somewhere (maybe you can have a small server for this?) and then have a custom builder plugin that could be used to trigger a deploy for your app by building a slim docker container that downloads the WAR file and executes it. If this is interesting to you but you don't feel you have the knowledge to do so, [consider sponsoring the work](https://github.com/dokku/.github/blob/master/SPONSORING.md).

Aside, I don't think I've heard of WAR files in like... 15+ years. The tech was great.
